cmake/IWYU.cmake

```diff
@@ -0,0 +1,40 @@
+find_program(IWYU_PATH NAMES iwyu_tool iwyu-tool)
+
+if(IWYU_PATH)
+    # iwyu is available
+    message(STATUS "Found IWYU: ${IWYU_PATH}")
+
+    # add a target to inspect number of times headers are included.
+    add_custom_target(iwyu-list-includes
+        COMMAND grep -hor "\"#include .*\""  ${PROJECT_SOURCE_DIR}/src | sort | uniq -c | sort -n
+    )
+
+    # find the standard library directory
+    execute_process(COMMAND clang -print-resource-dir
```

Comment:
Might be a good idea to first check via `find_program(CLANG_PATH clang)` if clang is installed on the system and use `${CLANG_PATH}` in this command.